File: flask/app/routes/ums/Permission.py
from flask import Blueprint, request
from app.services.ums.PermissionService import *
from app.utils.response import *

permission_bp = Blueprint("permission", __name__, url_prefix="/permission")

# 添加
@permission_bp.route("/add", methods=["POST"])
def add_permission():
    data = request.json
    return create_permission(data)

# 修改
@permission_bp.route("/edit/<int:permission_id>", methods=["POST"])
def edit_permission(permission_id):
    data = request.json
    return update_permission(permission_id, data)

# 删除
@permission_bp.route("/delete/<int:permission_id>", methods=["POST"])
def remove_permission(permission_id):
    return delete_permission(permission_id)

# 分页查询
@permission_bp.route("/list", methods=["GET"])
def list_permission():
    keyword = request.args.get("keyword", ' ')
    page = request.args.get("page", 1, type=int)
    page_size = request.args.get("page_size", 6, type=int)
    return list_permissions( keyword, page, page_size)

File: vue/src/api/ums/permission.js
import request from '@/utils/request'

// 获取权限列表
export function getPermissionList(keyword, page, page_size) {
  return request({
    url: '/permission/list',
    method: 'get',
    params: { keyword, page, page_size }
  })
}

// 添加权限
export function addPermission(data) {
  return request({
    url: '/permission/add',
    method: 'post',
    data
  })
}

// 更新权限
export function updatePermission(permission_id, data) {
  return request({
    url: `/permission/edit/${permission_id}`,
    method: 'post',
    data
  })
}

// 删除权限
export function deletePermission(permission_id) {
  return request({
    url: `/permission/delete/${permission_id}`,
    method: 'post'
  })
}
